### ISSUE TITLE
Multi-Release JAR support

### ISSUE THREAD

**Issue by gilles-duboscq** [Guardsquare/proguard#100]
Currently proguard doesn't support [Multi-Release JAR files](https://openjdk.java.net/jeps/238) and it actively filters out `META-INF/versions/**`.

When processing libraries, the recommended workaround](https://sourceforge.net/p/proguard/bugs/671/) is to find or crate a jar that contains the class files for the desired version.

However that doesn't help when the output of proguard should itself be a multi-release JAR.

**Comment by mrjameshamilton** [comment]
Hi @gilles-duboscq ! Thanks for your report. Indeed this is not yet supported but I'll tag this as `enhancement` and hopefully it can be added at some point.

**Comment by vjay82** [comment]
This feature would be much appreciated. I read somewhere that it was not implemented because of the enormous amount of work/complexity it would impose.
The current workaround is to remove all redundant/versioned files. I have a proposal for another workaround:
Most libraries only have a very little amount of versioned files, which are mostly used to work around JVM issues (like the switch between 8 and 9). Why don't you leave everything around these files untouched? I mean the originals and their versions. Maybe shrink names but even without that you would still achieve an enormous overall compression without breaking libs.
What I am doing now with libs, for example with Log4J2, is, to check the META-INF/versions. Build exclusions for the classes and copy the versioned files back in as a manual step. That is pretty tedious and error prone. I would gladly exchange a JAR with a few bytes more to get rid of that workflow...

**Comment by gilles-duboscq** [comment]
Another very tedious way to work around it might be:
* create a "java 8" jar by removing all `META-INF/versions` files, process that with proguard, save the mapping
* create a jar containing just the files in the next version (say `META-INF/versions/9`), copy the

**Comment by Ristovski** [comment]
I agree with @vjay82, there should be an option to simply copy `META-INF/versions/**` over to the processed jar.
FWIW, I am running into the same issue with log4j as well.

**Comment by badmannersteam** [comment]
I also ran into this problem when tried to change the logging level in runtime in Log4j2 in Proguard processed build of Compose app for Desktop, because some Log4j2 classes implementations for Java 9+ were not copied from source jar `/META-INF/versions/9`.

If someone met this issue too - here is a workaround for Compose `AbstractProguardTask`:
```kotlin
compose.desktop.application.buildTypes.release.proguard {
    configurationFiles.from(project.file("proguard-rules.pro"))
    obfuscate = true
    joinOutputJars = true
}

tasks.withType<AbstractProguardTask> {
    doLast {
        val log4jJar = project.configurations.runtimeClasspath.get().files
            .first { it.name.startsWith("log4j-api") }
            .toPath()

        val targetJar = destinationDir.file(mainJar.get().asFile.name).get().asFile.toPath()

        FileSystems.newFileSystem(log4jJar).use { log4jJarFs ->
            FileSystems.newFileSystem(targetJar).use { targetJarFs ->
                val sourceDir = log4jJarFs.getPath("/META-INF/versions/9")
                val targetDir = targetJarFs.getPath("/")

                Files.walk(sourceDir)
                    .filter { Files.isRegularFile(it) }
                    .forEach { file ->
                        val targetFile = targetDir.resolve(sourceDir.relativize(file))
                        Files.copy(file, targetFile, StandardCopyOption.REPLACE_EXISTING)
                    }
            }
        }
    }
}
```
